src/specialized.jl

```diff
@@ -58,7 +58,7 @@ function predict_string(ps::MemoryParserState)
             if ps.utf8data[s] == LATIN_U  # Unicode escape
                 t = ps.s
                 ps.s = s + 1
-                len += length(string(read_unicode_escape!(ps)).data)
+                len += sizeof(string(read_unicode_escape!(ps)))
```

Comment:
This could certainly be made faster, since it is a bit silly to allocate a string just to compute its size.

```jl
utf8size(c::Char) = utf8size(UInt32(c))
utf8size(c::UInt32) = c < 0x80 ? 1 : c < 0x800 ? 2 : c < 0x10000 ? 3 : c < 0x110000 ? 4 : 3
```